src/test/java/io/github/externschool/planner/service/UserServiceImplTest.java

```diff
@@ -0,0 +1,36 @@
+package io.github.externschool.planner.service;
+
+import io.github.externschool.planner.entity.User;
+import io.github.externschool.planner.repository.UserRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class UserServiceImplTest {
+
+    @Autowired
+    UserService userService;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Test
+    public void createNewUserTest(){
```

Comment:
This test checks does mail is saved, so the name can be "shouldSaveMailWhenMailIsCorrect".

Reply:
You can use Mocks, verify, test when null.

Reply:
I am creating user with two fields: id and name(email is the user's name) and yes, password field is null.  Service method creates real User.. Sorry, can't get what is incorrect ? Should I add password field?

Reply:
assertEquals(dmytro, retrieveDmytro)

Reply:
where are we getting null ? As I understand the user data does not come from outside, but comes from signup form. And if some data is incorrect the jdbc transaction will be rolled back by "autocommit" by default or submission form will ask you to try again )
Need help to clarify issue.